python/socketioclient/socketio_client.py

```diff
@@ -0,0 +1,115 @@
+import argparse
+import base64
+import hashlib
+import hmac
+import ssl
+import time
+import uuid
+import logzero
+import socketio
+
+logger = logzero.logger
+
+ctx = ssl.create_default_context()
+ctx.set_ciphers('DEFAULT')
+
+URL = 'https://stream.falconx.io'
+API_KEY = '<Your API_KEY>'
+PASSPHRASE = '<Your PASSPHRASE>'
+SECRET_KEY = '<Your SECRET_KEY>'
+
+def create_header(api_key, secret_key, passphrase):
+    timestamp = str(time.time())
+    message = timestamp + 'GET' + "/socket.io/"
+    hmac_key = base64.b64decode(secret_key)
+    signature = hmac.new(hmac_key, message.encode(), hashlib.sha256)
+    signature_b64 = base64.b64encode(signature.digest())
+    return {
+        'FX-ACCESS-SIGN': signature_b64.decode("utf-8"),
+        'FX-ACCESS-TIMESTAMP': timestamp,
+        'FX-ACCESS-KEY': api_key,
+        'FX-ACCESS-PASSPHRASE': passphrase,
+        'Content-Type': 'application/json',
+    }
+
+
+class FastRFSClient(socketio.ClientNamespace):
+    def __init__(self, namespace):
+        self.subscription_requests = []
+        self.previous_response_times = {}
+        self.running_avg_times = {}
+        super().__init__(namespace)
+
+    def populate_subscription_requests(self, token_pairs: list, levels: list):
+        for token_pair in token_pairs:
+            base_token, quote_token = token_pair.split("/")
+            self.subscription_requests.append({
+                'token_pair': {
+                    'base_token': base_token,
+                    'quote_token': quote_token
+                },
+                'quantity': levels,
+                'quantity_token': quote_token,  # only for v2/subscribe (optional)
+                'client_request_id': str(uuid.uuid4()),
+                'echo_id': True
+            })
+
+    def on_connect(self):
+        logger.info('Server connected.')
+        for subscription_request in self.subscription_requests:
+            self.emit('subscribe', subscription_request, namespace='/streaming')
+            client_request_id = subscription_request['client_request_id']
+            self.previous_response_times[client_request_id] = time.time()
+        logger.info('Finished subscribing.')
+
+    def on_disconnect(self, *args):
+        logger.info('Server disconnected.' + str(args))
+
+    def on_connect_error(self, *args):
+        logger.info("Cannot connect to the server. Error", args)
+
+    def on_response(self, *args):
+        response_time = time.time()
```

Comment:
hey, lets not do any of this, just a simple log statement should be enough

Reply:
Changed.